sdk/core/System.ClientModel/src/Internal/Logging/PipelineMessageLogger.cs

```diff
@@ -0,0 +1,258 @@
+                {
+                    ResponseContentTextBlock(_logger, requestId, blockNumber, textEncoding.GetString(content));
+                }
+                else
+                {
+                    ResponseContentBlock(_logger, requestId, blockNumber, content);
+                }
+            }
+        }
+        else
+        {
+            ClientModelEventSource.Log.ResponseContentBlock(requestId, blockNumber, content, textEncoding);
+        }
+    }
+
+    [LoggerMessage(LoggingEventIds.ResponseContentBlockEvent, LogLevel.Debug, "Response [{requestId}] content block {blockNumber}: {content}", SkipEnabledCheck = true, EventName = "ResponseContentBlock")]
+    private static partial void ResponseContentBlock(ILogger logger, string requestId, int blockNumber, byte[] content);
+
+    [LoggerMessage(LoggingEventIds.ResponseContentTextBlockEvent, LogLevel.Debug, "Response [{requestId}] content block {blockNumber}: {content}", SkipEnabledCheck = true, EventName = "ResponseContentTextBlock")]
+    private static partial void ResponseContentTextBlock(ILogger logger, string requestId, int blockNumber, string content);
+
+    #endregion
+
+    #region Error Response
+
+    public void LogErrorResponse(string requestId, PipelineResponse response, double seconds)
+    {
+        if (_logger is not null)
+        {
+            if (_logger.IsEnabled(LogLevel.Warning))
+            {
+                ErrorResponse(_logger, requestId, response.Status, response.ReasonPhrase, FormatHeaders(response.Headers), seconds);
+            }
+        }
+        else
+        {
+            ClientModelEventSource.Log.ErrorResponse(requestId, response, seconds, _sanitizer);
+        }
+    }
+
+    [LoggerMessage(LoggingEventIds.ErrorResponseEvent, LogLevel.Warning, "Error response [{requestId}] {status} {reasonPhrase} ({seconds:00.0}s)\r\n{headers}", SkipEnabledCheck = true, EventName = "ErrorResponse")]
+    private static partial void ErrorResponse(ILogger logger, string requestId, int status, string reasonPhrase, string headers, double seconds);
+
+    public void LogErrorResponseContent(string requestId, byte[] content, Encoding? textEncoding)
+    {
+        if (_logger is not null)
+        {
+            if (_logger.IsEnabled(LogLevel.Information))
+            {
+                if (textEncoding != null)
+                {
+                    ErrorResponseContentText(_logger, requestId, textEncoding.GetString(content));
+                }
+                else
+                {
+                    ErrorResponseContent(_logger, requestId, content);
+                }
+            }
+        }
+        else
+        {
+            ClientModelEventSource.Log.ErrorResponseContent(requestId, content, textEncoding);
+        }
+    }
+
+    [LoggerMessage(LoggingEventIds.ErrorResponseContentEvent, LogLevel.Information, "Error response [{requestId}] content: {content}", SkipEnabledCheck = true, EventName = "ErrorResponseContent")]
```

Comment:
Ilogger logs are super-easy to enrich with W3C trace-context, see https://learn.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-9.0#automatically-log-scope-with-spanid-traceid-parentid-baggage-and-tags

If you want to guarantee some correlation information is stamped by default, let's use `Activity.Current?.Id` as a `requestId` instead of inventing new standard.

If you want to support user-provided request-id, a few thought:
- why? do we have to?
- let's let them provide it via `Activity` - it's easy
-  if it's a must, let's only use it when they explicitly provided one - let's never generate ours and use `Activity.Current?.Id` to identify requests.

Reply:
The only reason this is here is because it's following the same behavior as the Event Source code, which is following the Azure.Core implementation. However, since ILogger isn't in Azure.Core I'll change this, thanks for the information about it!